src/DeserializerFactory.php

```diff
@@ -34,9 +35,25 @@ class DeserializerFactory {
 	 */
 	private $currentFactory;
 
-	public function __construct( Deserializer $dataValueDeserializer, EntityIdParser $idParser ) {
+	/**
+	 * @var DispatchableDeserializer|null $currentEntityDeserializer
+	 */
+	private $currentEntityDeserializer;
+
+	/**
+	 * @param Deserializer $dataValueDeserializer
+	 * @param EntityIdParser $idParser
+	 * @param DispatchableDeserializer|null $currentEntityDeserializer used instead of constructing
+	 *        a new current Deserializer for entities using a current DeserializerFactory.
+	 */
+	public function __construct(
+		Deserializer $dataValueDeserializer,
+		EntityIdParser $idParser,
+		DispatchableDeserializer $currentEntityDeserializer = null
+	) {
 		$this->legacyFactory = new LegacyDeserializerFactory( $dataValueDeserializer, $idParser );
 		$this->currentFactory = new CurrentDeserializerFactory( $dataValueDeserializer, $idParser );
```

Comment:
If $currentEntityDeserializer is given, $this->currentFactory should never be used, right? So perhaps set it to null.

Alternatively, we could have a trivial DeserializerFactory that just always returns the same serializer. Then we could just do   $this->currentFactory = new SingleDeserializerFactory( $currentEntityDeserializer ). 

Or we could have two implementations of DeserializerFactory - one that takes a factory for the current serializers, and one that takes a serializer. 

It all seems a bit silly. Just provide a way to get the legacy serializers, and let the caller sort it out.